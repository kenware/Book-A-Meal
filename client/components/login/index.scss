#login-bg {
    background: url('../../image/bg5.jpg') center center fixed; 
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
}
.lg-btn {
    margin-left: 7%;
    width: 70px;
    height: 30px;
    font-size: 20px;    
}
