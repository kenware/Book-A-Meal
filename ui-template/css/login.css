body{
  background-image: url("../image/ac.jpg");
}


.login-container{
  display: flex;
  height: auto;
}
.login-col{
  flex:1;
}
.login-wrapper{
    -moz-box-shadow: 15px 5px 5px 5px #272626;
    -webkit-box-shadow: 15px 5px 5px 5px #dfd5d5;
    box-shadow: 5px 5px 5px 5px #6e6060;
    background-color: white;
    margin-top: 200px;
    height: 200px;
    align-items:center;
   flex:3;
   height: 300px;;
}
.form-row{
    display: flex;
    padding: 10px 0;
}
.form-row label{
 padding: 10px 10px 10px 10px;
 flex:1;
}
.form-row input{
  flex:2;
  padding: 10px 10px 10px 10px;
  margin-right: 10px;
 }
 .form-row button{
  flex:2;
  padding: 10px 10px 10px 10px;
  margin-right: 10px;
  margin-left: 10px;
  

 }