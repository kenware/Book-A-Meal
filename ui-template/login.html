<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8" />
        <title>Book-A-Meal</title>
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <link rel="stylesheet" href="css/index.css">
        <link rel="stylesheet" href="css/login.css">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
        <script src="js/index.js"></script>
    </head>
    <body>
        <div class="nav" id="myTopnav" style="position: fixed;width:100%;top:0;">
            <a href="index.html" class="active" style="font-size:25px;">Book-A-Meal</a>
            <a href="dashboard.html">Dashboard</a>
            <a href="admin.html">Admin</a>
            <a href="admin-set-menu.html">Set-Menu</a>
            <a href="admin-all.html">All-Menu</a>
            <a href="login.html">Login</a>
            <a href="register.html">SignUp</a>
          
            <a href="index.html" class="active" style="font-size:23px;">Home</a>
            <a href="javascript:void(0);" style="font-size:15px;" class="icon" onclick="myFunction()"><span class="fa fa-bars fa-2x"></span></a>
        </div>
        <div class=login-container>
            <div class="login-col"></div>
            <div class="login-wrapper">
                <h2 class'login-header'>Enter your details to logIn</h2> 
                <form action="dashboard.html">
                    <div class="form-row">
                        <label for="name">Username</label>
                        <input type="text" id="name">
                    </div>
                    <div class="form-row">
                        <label for="password">Password</label>
                        <input type="password" id="password">
                    </div>
                    <div class="form-row">
                        <button type="submit" class="button lg"> Login</button>
                    </div>
                    <div class="form-row">
                    <label for="inputPassword3"><a href="/signup">signup</a></label>
                    <label for="inputPassword3"><a href="/signup">forgot password?</a></label>
                    </div>
                </form>
            </div> 
            <div class="login-col"></div>
            
        </div>
        <footer class="footer" style="margin-top:30px;">
            <div class="footer-col">
                <h2 class="">
                    Book-A-Meal
                </h2>
                <ul>
                    <li> <a href="login.html" class="text-white">LogIn</a></li>
                    <li> <br><a href="register.html" class="text-white">SignUp</a></li>
                    <li><br><a href="dashboard.html" class="text-white">Dashboard</a></li>
                    <li><br><a href="#" class="text-white">Get In Touch</a></li>
                </ul>
            </div>
            <div class="footer-col">
                <h2 class="">
                    Recent News
                </h2>
                <p>
                    Book-a-Meal app is designed to make live better
                    and easier. You can get your favorite meal with just
                    a click any where around the world.
                    <br>
                    <br> Dont hesistate to make
                    your order now.We are available 24/7 
                </p>
            </div>
            <div class="footer-col">
                <h2 class="">
                    SUBSCRIBE
                </h2>
                <p>
                    Get Monthly Updates on our Meal and Recipes and
                    also free resources.
                </p>
                <div class="form-row">
                        <input type="email"  placeholder="Email" style="background-color:white;">
                </div>
                <div class="form-row">
                    <button href="" class="button" type="submit" style="margin-left:25px;">Subscribe</button>
                </div>
            </div>
            <div class="footer-col social">
                <h2 class="">
                    Connect With Us
                </h2>
                <p>
                    Get in touch with us on various social media
                </p>
                <span class="fa fa-twitter"></span>
                <span class="fa fa-facebook"></span>
                <span class="fa fa-likedIn"></span>
                <span class="fa fa-twitter"></span>
            </div>
        </footer>
        <div class="footer-row">
            <p class="">
                Â© Copyright 2018 Book-A-Meal - All Rights Reserved
            </p>
        </div> 
    </body>
</html>
