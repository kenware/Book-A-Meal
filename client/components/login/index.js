import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import Header from '../header/index';
import Footer from '../footer/index';
import './index.scss';
import * as actions from '../../redux/Action/action';

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      login: 'Login',
      username: '',
      password: '',
      userField: '',
      passField: ''
    };
    this.onChange = this.onChange.bind(this);
    this.login = this.login.bind(this);
  }
  /**
   * ifecycle hook called when component is receives props
   * @param  {} props if new props from redux with error arrives
   * change the state login button from loading icon to
   * to text
   */
  componentWillReceiveProps(props) {
    if (props.errorMessage.loginError) {
      toast.error('Wrong login credentials', {
        className: 'toasterror'
      });
      this.setState({ login: 'Login' });
    }
  }
  /**
   * remove error props in redux store if component unmount
   */
  componentWillUnmount() {
    this.props.actions.clearMessages();
  }
  /**
   * @param  {} e set state on input onchange event
   */
  onChange(e) {
    const { state } = this;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }
  /**
   * @param  {} e login method
   */
  login(e) {
    e.preventDefault();
    const {
      password,
      username
    } = this.state;
    if (!username) { this.setState({ userField: 'Username is required' }); return; }
    if (!password) { this.setState({ passField: 'Password is required' }); return; }
    this.props.actions.login(username, password);
    this.setState({
      login: (<div><i className="fa fa-spinner fa-spin fa-2x fa-fw" aria-hidden="true" /></div>),
      userField: '',
      passField: ''
    });
    this.props.actions.clearMessages();
  }
  render() {
    return (
      <div>
        <Header />
        <ToastContainer autoClose={8000} />
        <div className="register-container" id="login-bg">
          <div className="register-wrapper" style={{ margin: '100px 25% 100px 25%' }}>
            <h3 className="text-center danger">{this.props.errorMessage.authError }</h3>
            <form action="dashboard.html" className="login">
              <div className="form-field">
                <span className="text-center danger">{this.state.userField }</span>
                <input onChange={this.onChange} type="text" id="username" name="username" placeholder="Username" autoComplete="true" required />
              </div>
              <div className="form-field">
                <span className="text-center danger">{this.state.passField }</span>
                <input onChange={this.onChange} type="password" name="password" id="password" placeholder="Password" autoComplete="true" required />
              </div> <br />
              <div style={{ display: 'flex' }}>
                <button name="submit" type="submit" className="button lg-btn" onClick={this.login}>{this.state.login} </button>
              </div> <br />
              <div className="form-field">
                <Link to="/register">SignUp</Link>&nbsp;&nbsp;&nbsp;&nbsp;
                <Link to="/resetLink">forgot password? </Link>
              </div>
            </form>

          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
Login.propTypes = {
  errorMessage: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};
export function mapStateToProps(state) {
  return {
    errorMessage: state.errorMessage
  };
}
export function mapDispatchToProps(dispatch) {
  return { actions: bindActionCreators(actions, dispatch) };
}
export default connect(mapStateToProps, mapDispatchToProps)(Login);
