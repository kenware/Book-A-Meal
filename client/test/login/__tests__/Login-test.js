import React from 'react';
import { shallow } from 'enzyme';
import { Login, mapDispatchToProps, mapStateToProps } from '../../../components/login/index';
import { props, emptyProps } from '../loginMock';

describe('Test Login Component', () => {
  it('renders correctly', () => {
    const tree = shallow(<Login {...emptyProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should have one login container class', () => {
    const tree = shallow(<Login {...props} />);
    expect(tree.find('.register-container')).toHaveLength(1);
  });

  it('should have one login container class', () => {
    const componentWillUnmount = jest.spyOn(Login.prototype, 'componentWillUnmount');
    const tree = shallow(<Login {...props} />);
    const wrapper = tree.instance();
    wrapper.componentWillUnmount();
    expect(componentWillUnmount).toHaveBeenCalled();
  });

  it('should respond to change event and change the state of the Login Component', () => {
    const login = jest.spyOn(Login.prototype, 'login');
    const tree = shallow(<Login {...props} />);
    const wrapper = tree.instance();
    wrapper.login({ preventDefault: jest.fn() });
    tree.find('#username').simulate('change', { target: { name: 'username', value: 'kevin' } });
    wrapper.login({ preventDefault: jest.fn() });
    tree.find('#password').simulate('change', { target: { name: 'password', value: 'cats' } });
    wrapper.login({ preventDefault: jest.fn() });
    expect(tree).toMatchSnapshot();
    expect(login).toHaveBeenCalled();
  });

  it('should respond to mapStateToProps methods', () => {
    const ownProps = { match: { params: { mealId: 1 } } };
    const tree = mapStateToProps(props, ownProps);
    expect(tree).toMatchSnapshot();
  });

  it('should respond to mapDispatchToProps methods', () => {
    const tree = mapDispatchToProps(emptyProps);
    expect(tree).toMatchSnapshot();
  });

  it('should respond to lifecycle methods', () => {
    const componentWillReceiveProps = jest.spyOn(Login.prototype, 'componentWillReceiveProps');
    const tree = shallow(<Login {...props} />);
    tree.instance().componentWillReceiveProps({ errorMessage: { loginError: 'error during login' } });
    expect(tree).toMatchSnapshot();
    expect(componentWillReceiveProps).toHaveBeenCalled();
  });
});
