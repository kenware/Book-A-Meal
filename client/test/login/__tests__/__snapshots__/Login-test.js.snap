// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login Component renders correctly 1`] = `
<div>
  <Nav />
  <div
    className="register-container"
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
    >
      <h2
        className="login-header"
      >
        Enter your details to Login
      </h2>
      <br />
      <h3
        className="text-center danger"
      />
      <h3
        className="text-center danger"
      />
      <form
        action="dashboard.html"
        className="login"
      >
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Username 
            <br />
            <span
              className="text-center danger"
            />
          </label>
          <input
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Keny"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="password"
          >
            Password 
            <br />
            <span
              className="text-center danger"
            />
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
          />
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg"
            name="submit"
            onClick={[Function]}
            type="submit"
          >
            Login
          </button>
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="inputPassword3"
          >
            <Link
              replace={false}
              to="/register"
            >
              SignUp
            </Link>
                
            <Link
              replace={false}
              to="/resetLink"
            >
              forgot password? 
            </Link>
          </label>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
  <footer />
</div>
`;

exports[`Test Login Component should recieve error message props with wrong credentials 1`] = `
<div>
  <Nav />
  <div
    className="register-container"
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
    >
      <h2
        className="login-header"
      >
        Enter your details to Login
      </h2>
      <br />
      <h3
        className="text-center danger"
      />
      <h3
        className="text-center danger"
      />
      <form
        action="dashboard.html"
        className="login"
      >
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Username 
            <br />
            <span
              className="text-center danger"
            />
          </label>
          <input
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Keny"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="password"
          >
            Password 
            <br />
            <span
              className="text-center danger"
            />
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
          />
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg"
            name="submit"
            onClick={[Function]}
            type="submit"
          >
            Login
          </button>
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="inputPassword3"
          >
            <Link
              replace={false}
              to="/register"
            >
              SignUp
            </Link>
                
            <Link
              replace={false}
              to="/resetLink"
            >
              forgot password? 
            </Link>
          </label>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
  <footer />
</div>
`;

exports[`Test Login Component should recieve error message props with wrong credentials 2`] = `
<div>
  <Nav />
  <div
    className="register-container"
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
    >
      <h2
        className="login-header"
      >
        Enter your details to Login
      </h2>
      <br />
      <h3
        className="text-center danger"
      >
        Wrong credentials
      </h3>
      <h3
        className="text-center danger"
      />
      <form
        action="dashboard.html"
        className="login"
      >
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Username 
            <br />
            <span
              className="text-center danger"
            />
          </label>
          <input
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Keny"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="password"
          >
            Password 
            <br />
            <span
              className="text-center danger"
            />
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
          />
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg"
            name="submit"
            onClick={[Function]}
            type="submit"
          >
            Login
          </button>
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="inputPassword3"
          >
            <Link
              replace={false}
              to="/register"
            >
              SignUp
            </Link>
                
            <Link
              replace={false}
              to="/resetLink"
            >
              forgot password? 
            </Link>
          </label>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
  <footer />
</div>
`;

exports[`Test Login Component should respond to change event and change the state of the Login Component 1`] = `
<div>
  <Nav />
  <div
    className="register-container"
  >
    <div
      className="register-col"
    />
    <div
      className="register-wrapper"
    >
      <h2
        className="login-header"
      >
        Enter your details to Login
      </h2>
      <br />
      <h3
        className="text-center danger"
      >
        Wrong credentials
      </h3>
      <h3
        className="text-center danger"
      />
      <form
        action="dashboard.html"
        className="login"
      >
        <div
          className="form-field"
        >
          <label
            htmlFor="name"
          >
            Username 
            <br />
            <span
              className="text-center danger"
            >
              Username is required
            </span>
          </label>
          <input
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Keny"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="password"
          >
            Password 
            <br />
            <span
              className="text-center danger"
            >
              Password is required
            </span>
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
          />
        </div>
        <div
          className="form-field"
        >
          <button
            className="button lg"
            name="submit"
            onClick={[Function]}
            type="submit"
          >
            Login
          </button>
        </div>
        <div
          className="form-field"
        >
          <label
            htmlFor="inputPassword3"
          >
            <Link
              replace={false}
              to="/register"
            >
              SignUp
            </Link>
                
            <Link
              replace={false}
              to="/resetLink"
            >
              forgot password? 
            </Link>
          </label>
        </div>
      </form>
    </div>
    <div
      className="register-col"
    />
  </div>
  <footer />
</div>
`;

exports[`Test Login Component should respond to mapDispatchToProps methods 1`] = `
Object {
  "actions": Object {
    "clearMessages": [Function],
    "getNotifications": [Function],
    "getUser": [Function],
    "loadErrorMessage": [Function],
    "loadMostOrdered": [Function],
    "loadMostOrderedMeal": [Function],
    "loadNotifications": [Function],
    "loadSuccessMessage": [Function],
    "loadUser": [Function],
    "login": [Function],
    "refreshToken": [Function],
    "register": [Function],
    "resetLink": [Function],
    "updateProfile": [Function],
    "upgrade": [Function],
  },
}
`;

exports[`Test Login Component should respond to mapStateToProps methods 1`] = `
Object {
  "errorMessage": Object {
    "loginError": "Wrong credentials",
  },
}
`;
